src/serialization/sb2.js

```diff
@@ -423,9 +444,54 @@ const parseScratchObject = function (object, runtime, extensions, topLevel, zip)
         }
     }
 
+    // If included, parse any and all comments on the object (this includes top-level
+    // workspace comments as well as comments attached to specific blocks)
+    const blockComments = {};
+    if (object.hasOwnProperty('scriptComments')) {
+        const comments = object.scriptComments.map(commentDesc => {
+            const isBlockComment = commentDesc[5] >= 0;
```

Comment:
It might be nice to destructure the `commentDesc` here, and use that as place to condense the documentation of what each field is, e.g.

```
const [
    xPosition, 
    yPosition,
    width,
    height,
    expanded,
    flattenedBlockIndex,
    textContent
] = commentDesc;